weblate/screenshots/models.py

```diff
@@ -76,3 +89,49 @@ def change_screenshot_assignment(sender, instance, action, **kwargs):
         name="UnusedScreenshot"
     ).exists():
         component_alerts.delay([instance.pk])
+
+
+@receiver(vcs_post_update)
+def update_screenshots_from_repo(sender, component, previous_head: str, **kwargs):
+    repository = component.repository
+    changed_files = repository.get_changed_files(compare_to=previous_head)
+    screenshots = Screenshot.objects.filter(
+        translation__component=component, repository_filename__in=changed_files
+    )
+
+    # Update existing screenshots
+    for screenshot in screenshots:
+        filename = screenshot.repository_filename
+        changed_files.remove(filename)
+        current_screenshot_object = screenshot.image
+        new_screenshot_image = repository.get_file(filename, previous_head)
+
+        try:
+            validate_bitmap(new_screenshot_image)
+        except ValidationError:
+            report_error(cause="Failed to validate image from repository")
+            continue
+
+        current_screenshot_object.image = new_screenshot_image
+        current_screenshot_object.save(update_fields=["image"])
```

Comment:
Does this really update the image? I'd rather expect some operation with `File` object to be needed, see https://docs.djangoproject.com/en/4.1/topics/files/. Similar applies for creating the screenshot lower.